CMake/ITKSetStandardCompilerFlags.cmake

```diff
@@ -130,14 +130,62 @@ function(check_compiler_warning_flags c_warning_flags_var cxx_warning_flags_var)
     ## Clang compiler likes to warn about this feature that is technically only in
     ## c++0x, but works on many compilers, and if it fails, then alternate methods are used
 
-  check_c_compiler_warning_flags(CMAKE_C_WARNING_FLAGS ${c_flags} ${c_and_cxx_flags})
-  check_cxx_compiler_warning_flags(CMAKE_CXX_WARNING_FLAGS ${c_and_cxx_flags} ${cxx_flags})
+  check_c_compiler_flags(CMAKE_C_WARNING_FLAGS ${c_flags} ${c_and_cxx_flags})
+  check_cxx_compiler_flags(CMAKE_CXX_WARNING_FLAGS ${c_and_cxx_flags} ${cxx_flags})
 
   set(${c_warning_flags_var} "${CMAKE_C_WARNING_FLAGS}" PARENT_SCOPE)
   set(${cxx_warning_flags_var} "${CMAKE_CXX_WARNING_FLAGS}" PARENT_SCOPE)
 endfunction()
 
 
+function(check_compiler_optimization_flags c_optimization_flags_var cxx_optimization_flags_var)
+  set(${c_optimization_flags_var} "" PARENT_SCOPE)
+  set(${cxx_optimization_flags_var} "" PARENT_SCOPE)
+
+  if(MSVC)
+      set(InstructionSetOptimizationFlags /arch:SSE /arch:SSE2 /arch:/AVX ) # /arch:/AVX2
+      set(c_and_cxx_flags
+         # https://docs.microsoft.com/en-us/cpp/build/reference/arch-x64?view=vs-2015
+         ${InstructionSetOptimizationFlags}
+         /arch:SSE /arch:SSE2 /arch:/AVX # Circa 2008 CPU's use /arch:/AVX2 for circa 2013 CPU support
+         )
+  else()
+    if (${CMAKE_C_COMPILER} MATCHES "icc.*$")
+      set(USING_INTEL_ICC_COMPILER TRUE)
+    endif()
+    if (${CMAKE_CXX_COMPILER} MATCHES "icpc.*$")
+      set(USING_INTEL_ICC_COMPILER TRUE)
+    endif()
+    if(USING_INTEL_ICC_COMPILER)
+      set(InstructionSetOptimizationFlags "" )
+    else()
+      set(InstructionSetOptimizationFlags "")
+    endif ()
+
+  # Check this list on C compiler only
+  set(c_flags "")
+
+  # Check this list on C++ compiler only
+  set(cxx_flags "")
+
+  # Check this list on both C and C++ compilers
+  set(c_and_cxx_flags
+     # https://gcc.gnu.org/onlinedocs/gcc-4.8.0/gcc/i386-and-x86_002d64-Options.html
+     # NOTE the corei7 release date was 2008
+     ${InstructionSetOptimizationFlags}
+     -mtune=native # Tune the code for the computer used compile ITK, but allow running on generic cpu archetectures
```

Comment:
Will this be ignored on Visual Studio?

Reply:
Line 152 is the else part, so Yes it will be ignored.  I fixed indentation of lines 165-177 to reflect the else indentation problem.